test/v2/connectors/RetrieveHistoricFhlUkPropertyAnnualSubmissionConnectorSpec.scala

```diff
@@ -0,0 +1,99 @@
+ */
+
+package v2.connectors
+
+import mocks.MockAppConfig
+import org.scalamock.handlers.CallHandler
+import v2.mocks.MockHttpClient
+import v2.models.domain.{ Nino, TaxYear }
+import v2.models.errors.{ DownstreamErrorCode, DownstreamErrors }
+import v2.models.outcomes.ResponseWrapper
+import v2.models.request.retrieveHistoricFhlUkPropertyAnnualSubmission.RetrieveHistoricFhlUkPropertyAnnualSubmissionRequest
+import v2.models.response.retrieveHistoricFhlUkPropertyAnnualSubmission.{
+  AnnualAdjustments,
+  AnnualAllowances,
+  RetrieveHistoricFhlUkPropertyAnnualSubmissionResponse
+}
+
+import scala.concurrent.Future
+
+class RetrieveHistoricFhlUkPropertyAnnualSubmissionConnectorSpec extends ConnectorSpec {
+
+  val nino: String              = "AA123456A"
+  val mtdTaxYear: String        = "2019-20"
+  val downstreamTaxYear: String = "2020"
+
+  val request: RetrieveHistoricFhlUkPropertyAnnualSubmissionRequest = RetrieveHistoricFhlUkPropertyAnnualSubmissionRequest(
+    nino = Nino(nino),
+    taxYear = TaxYear.fromMtd(mtdTaxYear)
+  )
+
+  val annualAdjustments = AnnualAdjustments(None, None, None, true, None, false, None)
+  val annualAllowances  = AnnualAllowances(None, None, None, None)
+
+  def responseWith(annualAdjustments: Option[AnnualAdjustments],
+                   annualAllowances: Option[AnnualAllowances]): RetrieveHistoricFhlUkPropertyAnnualSubmissionResponse =
+    RetrieveHistoricFhlUkPropertyAnnualSubmissionResponse(annualAdjustments, annualAllowances)
+
+  class Test extends MockHttpClient with MockAppConfig {
+
+    val connector: RetrieveHistoricFhlUkPropertyAnnualSubmissionConnector = new RetrieveHistoricFhlUkPropertyAnnualSubmissionConnector(
+      http = mockHttpClient,
+      appConfig = mockAppConfig
+    )
+
+    MockAppConfig.desBaseUrl returns baseUrl
+    MockAppConfig.desToken returns "des-token"
+    MockAppConfig.desEnvironment returns "des-environment"
+    MockAppConfig.desEnvironmentHeaders returns Some(allowedDownstreamHeaders)
+
+    def stubHttpResponse(outcome: DownstreamOutcome[RetrieveHistoricFhlUkPropertyAnnualSubmissionResponse])
+      : CallHandler[Future[DownstreamOutcome[RetrieveHistoricFhlUkPropertyAnnualSubmissionResponse]]]#Derived = {
+      MockHttpClient
+        .get(
+          url = s"$baseUrl/income-tax/nino/$nino/uk-properties/furnished-holiday-lettings/annual-summaries/$downstreamTaxYear",
+          config = dummyDesHeaderCarrierConfig,
+          requiredHeaders = requiredDesHeaders,
+        )
+        .returns(Future.successful(outcome))
+    }
+  }
+
+  "connector" when {
+    "request asks for a historic FHL UK property annual submission" must {
+      "return a valid result" in new Test {
+        val response = responseWith(Some(annualAdjustments), Some(annualAllowances))
+        val outcome  = Right(ResponseWrapper(correlationId, response))
+
+        stubHttpResponse(outcome)
+
+        val result = await(connector.retrieve(request))
+        result shouldBe outcome
```

Comment:
I like how you've structured this test, saving the responses to variables and having the final line be "result shouldBe outcome" makes it easier to read than some of the other tests in our codebase :)